app/workers/email_archive_worker.rb

```diff
@@ -4,17 +4,20 @@ class EmailArchiveWorker
   sidekiq_options queue: :cleanup
 
   LOCK_NAME = "email_archive_worker".freeze
+  BATCH_SIZE = 1000
 
   def perform
     return unless ENV.include?("EMAIL_ARCHIVE_S3_ENABLED")
     Email.with_advisory_lock(LOCK_NAME, timeout_seconds: 0) do
       start_time = Time.zone.now
       archived_count = 0
 
-      EmailArchiveQuery.call.in_batches do |batch|
-        Email.transaction do
-          archived_count += archive_batch(batch.as_json)
-        end
+      loop do
+        batch = EmailArchiveQuery.call.limit(BATCH_SIZE).as_json
```

Comment:
Something that would be nice to do (if it didn't impact performance too much) is to order this by finished_sending_at ASC as that would produce more consistent S3 files. 

Not a must have naturally since we didn't have it before when using `in_batches`

Reply:
> Something that would be nice to do (if it didn't impact performance too much) is to order this by finished_sending_at ASC as that would produce more consistent S3 files.

Ordering within the batch should be fine in terms of performance, but I don't know if that's useful. But I think that ordering overall is too expensive, this is just based on my trial query not yet finishing...

Reply:
Cool - yeah not necessary to order in the batch (that actually happens later at archive to s3 part if I remember correctly.